github/release_notes/util.py

```diff
@@ -486,14 +442,31 @@ def release_note_block(
     category = code_block.get('category')
     target_group = code_block.get('target_group')
     source_repo = code_block.get('source_repo')
-    if source_repo:
+
+    if source_component:
+        reference_id = code_block.get('reference_id')
+        reference_type = reference_type_for_type_identifier(code_block.get('reference_type'))
+        user_login = code_block.get('user')
+    elif source_repo:
+        try:
+            # try to fetch cd for the parsed source repo. The actual version does not matter,
+            # we're only interested in the components GithubAccess (we assume it does not
+            # change).
+            source_cd = product.v2.greatest_component_version_by_name(
+                component_name=source_repo,
+                ctx_repo_base_url=current_component.current_repository_ctx().baseUrl,
+                cache_dir=cache_dir,
+            )
+            source_component = source_cd.component
+        except requests.exceptions.HTTPError:
+            print(f'Unable to retrieve component descriptor for source repository {source_repo}')
+            return None
+
         reference_type = reference_type_for_type_identifier(code_block.get('reference_type'))
         reference_id = code_block.get('reference_id')
         user_login = code_block.get('user')
     else:
-        source_repo = cn_current_repo.name()
-        reference_type = reference_type
-        reference_id = reference_id
+        source_component = current_component
```

Comment:
source is used quite often in this piece of code. what is the relation between source and current?